src/org/voltdb/jsr107/CacheEventConsumer.java

```diff
@@ -148,10 +150,10 @@ public void run() {
 
             }
 
-            consumer.close();
-
         } catch (Exception e1) {
             CacheEventConsumer.msg(e1);
+        } finally {
+            consumer.close();
```

Comment:
Moved `close` to the `finally` section to be sure to close the consumer. In previous code in case of some errors (for example thrown from the `poll` method) we wouldn't close the consumer.

Reply:
{bless}